api/envoy/config/core/v3/protocol.proto

```diff
@@ -111,6 +111,10 @@ message QuicProtocolOptions {
     lte {seconds: 600}
     gte {seconds: 1}
   }];
+
+  // Maximum packet length for QUIC connections. It refers to the largest size of a QUIC packet that can be transmitted over the connection.
+  // If not specified, the default value defined in `QUICHE <https://github.com/google/quiche/blob/main/quiche/quic/core/quic_constants.h>` is used.
+  google.protobuf.UInt64Value max_packet_length = 9;
```

Comment:
Thanks for the comments, @RyanTheOptimist and @DavidSchinazi. I will send more commits soon but want to quickly check if we should set a reasonable bound for the max_packet_length option like other options in this config. One range I can think of is 1 <= `max_packet_length` <= 65,535 (max IP packet size), but I'm wondering if we should just not set the bound here. I used `UInt64Value` since the corresponding QUICHE variable and constants (e.g. https://github.com/google/quiche/blob/4249f8025caed1e3d71d04d9cadf42251acb7cac/quiche/quic/core/quic_constants.h#L32) uses `QuicByteCount`, which is `uint64_t` internally. Let me know if you have any thoughts on these.

Reply:
I don't think Envoy needs to put a cap on this value. Internally, QUICHE will cap it at 1452, but we shouldn't duplicate that constant or logic here

Reply:
SGTM. Let me know if you have any other comments on the new code.